src/systems/systems/src/pressurization/cabin_pressure_simulation.rs

```diff
@@ -0,0 +1,252 @@
+
+        let rate_of_change_for_delta = INTERNAL_FLOW_RATE_CHANGE * context.delta_as_secs_f64();
+
+        if packs_are_on {
+            if VolumeRate::new::<cubic_meter_per_second>(FLOW_RATE_WITH_PACKS_ON)
+                > self.cabin_flow_in
+            {
+                self.cabin_flow_in
+                    + VolumeRate::new::<cubic_meter_per_second>(rate_of_change_for_delta.min(
+                        FLOW_RATE_WITH_PACKS_ON
+                            - self.cabin_flow_in.get::<cubic_meter_per_second>(),
+                    ))
+            } else {
+                VolumeRate::new::<cubic_meter_per_second>(FLOW_RATE_WITH_PACKS_ON)
+            }
+        } else if self.cabin_flow_in > VolumeRate::new::<cubic_meter_per_second>(0.) {
+            self.cabin_flow_in
+                - VolumeRate::new::<cubic_meter_per_second>(
+                    rate_of_change_for_delta
+                        .min(self.cabin_flow_in.get::<cubic_meter_per_second>()),
+                )
+        } else {
+            VolumeRate::new::<cubic_meter_per_second>(0.)
+        }
+    }
+
+    fn calculate_cabin_flow_out(&self) -> VolumeRate {
+        let area_leakage = Self::AREA_LEAKAGE
+            + Self::SAFETY_VALVE_SIZE * self.safety_valve_open_amount.get::<ratio>();
+        VolumeRate::new::<cubic_meter_per_second>(
+            self.flow_coefficient
+                * area_leakage
+                * ((2.
+                    * (Self::GAMMA / (Self::GAMMA - 1.))
+                    * Self::RHO
+                    * self.cabin_pressure.get::<pascal>()
+                    * self.z_coefficient)
+                    .abs())
+                .sqrt(),
+        )
+    }
+
+    fn set_cabin_vs(&self) -> Velocity {
+        let vertical_speed = (self.outflow_valve_open_amount.get::<ratio>()
+            * Self::OFV_SIZE
+            * self.flow_coefficient
+            * ((2.
+                * (Self::GAMMA / (Self::GAMMA - 1.))
+                * Self::RHO
+                * self.cabin_pressure.get::<pascal>()
+                * self.z_coefficient)
+                .abs())
+            .sqrt()
+            - self.cabin_flow_in.get::<cubic_meter_per_second>()
+            + self.cabin_flow_out.get::<cubic_meter_per_second>())
+            / ((Self::RHO * Self::G * Self::CABIN_VOLUME) / (Self::R * Self::T_0));
+        Velocity::new::<meter_per_second>(vertical_speed)
+    }
+
+    fn calculate_cabin_pressure(
+        &self,
+        context: &UpdateContext,
+        lgciu_gear_compressed: bool,
+        simulation_is_ground: bool,
+    ) -> Pressure {
+        if simulation_is_ground && !lgciu_gear_compressed {
+            // Formula to simulate pressure start state if starting in flight
+            let ambient_pressure: f64 = context.ambient_pressure().get::<hectopascal>();
+            Pressure::new::<hectopascal>(
+                -0.0002 * ambient_pressure.powf(2.) + 0.5463 * ambient_pressure + 658.85,
+            )
```

Comment:
I'm not sure I understand this. What I'm thinking is: this code exists to set the system state when spawning in the air. However, are you sure that `simulation_is_ground` will be true in that case? My experience has usually been the opposite:  Even when spawning on ground, the simulator's own variables take a few moments to discover that it's on the ground, as the ground model loads in and the simulator initially believes it's hundreds of feet in the air.

On a grander scale (outside of the scope of this PR), this is not the first time that we're doing some weird things to deal with "spawn in situation X" cases. We should probably consider a generalized way (maybe using the SimulationVisitor) to deal with these special starts which we trigger once centrally, and then each system can decide how it wants to seed it's internal variables.

Reply:
Yes that's exactly it. This sets the cabin pressure when spawning in the air. I've just tried it a couple of times now and it does seem to work, but I agree it's not an elegant solution at all. I'm happy to change it to a more standard approach if you have any ideas on how to implement it, I'm afraid I would need some help on how to make that work 😃.

Reply:
As mentioned below to a different comment by @davidwalschots , I've changed this now to include an `initialized` field. This should make it clearer and more robust for now until we find a better way.